### PR TITLE
feat: added name wallet functionality to settings and connection screen

### PR DESCRIPTION
**Pull request by wadeking98**
# Summary of Changes

- Added the ability to change the wallet name from the settings screen
- Added the ability to change the wallet name from the connection screen
- Added functionality to update the agent label without restarting

https://github.com/hyperledger/aries-mobile-agent-react-native/assets/36937407/c8fe992f-4713-4dbf-bcec-d381302b6e7a

image

image

image

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

**Comment by bryce-mcmath** [comment]
What happens if you change your wallet name after connecting? Does the label at any point get updated in the other wallet's contacts?

**Comment by wadeking98** [comment]
> What happens if you change your wallet name after connecting? Does the label at any point get updated in the other wallet's contacts?

No, the connection label stays the same. Changing the wallet name will only change the label for future connections

**Comment by bryce-mcmath** [comment]
> No, the connection label stays the same. Changing the wallet name will only change the label for future connections

Ok, that's fine imo. Maybe just let Kim know. Also the position of the wallet name in the settings menu looks like it should be aligned to the left like the rest of the headings. Looking at this wireframe: https://xd.adobe.com/view/0900b5b9-22d7-44aa-b9ee-e3693e33a5f3-5c53/screen/bcaa6e11-910e-42dd-96b1-c9b582182594/

File: packages/legacy/core/App/screens/Settings.tsx [hunk 1]
```diff
@@ -238,12 +245,30 @@ const Settings: React.FC<SettingsProps> = ({ navigation }) => {
...
+  const SectionHeader: React.FC<{ icon: SettingIcon; iconRight?: SettingIcon; title: string }> = ({
+    icon,
+    iconRight,
+    title,
+  }) => (
+    <View style={[styles.section, styles.sectionHeader, { justifyContent: iconRight ? 'space-between' : undefined }]}>
+      <Icon
+        accessible={false}
+        name={icon.name}
+        size={icon.size ?? 24}
```
Comment (jleach) [inline packages/legacy/core/App/screens/Settings.tsx]: Consider defining `defaultIconSize` in just one place at the top of the file.

**Review by jleach — DISMISSED** [review]

File: packages/legacy/core/App/screens/Settings.tsx [hunk 2]
```diff
@@ -102,8 +102,15 @@ const Settings: React.FC<SettingsProps> = ({ navigation }) => {
   const settingsSections: SettingSection[] = [
     {
       header: {
-        icon: 'apartment',
-        title: t('Screens.Contacts'),
+        icon: { name: store.preferences.useConnectionInviterCapability ? 'person' : 'apartment', size: 30 },
+        title: store.preferences.useConnectionInviterCapability ? store.preferences.walletName : t('Screens.Contacts'),
+        iconRight: {
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/screens/Settings.tsx:107]: I am misreading this or is the edit icon not gated?

Comment (wadeking98) [reply packages/legacy/core/App/screens/Settings.tsx:107]: it is gated in the JSX template: https://github.com/hyperledger/aries-mobile-agent-react-native/blob/7be0b09278392c174b99091b0d3cef6e7c022df4/packages/legacy/core/App/screens/Settings.tsx#L254

**Review by bryce-mcmath — DISMISSED** [review]

**Review by bryce-mcmath — APPROVED** [review]

**Pull request merged**